app/lib/main.dart

```diff
@@ -10,28 +11,33 @@ class MyApp extends StatelessWidget {
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // TRY THIS: Try running your application with "flutter run". You'll see
-        // the application has a blue toolbar. Then, without quitting the app,
-        // try changing the seedColor in the colorScheme below to Colors.green
-        // and then invoke "hot reload" (save your changes or press the "hot
-        // reload" button in a Flutter-supported IDE, or press "r" if you used
-        // the command line to start the app).
-        //
-        // Notice that the counter didn't reset back to zero; the application
-        // state is not lost during the reload. To reset the state, use hot
-        // restart instead.
-        //
-        // This works for code too, not just values: Most code changes can be
-        // tested with just a hot reload.
-        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
-        useMaterial3: true,
+    return MultiProvider(
+      providers: [
+        ChangeNotifierProvider(create: (_) => Counter())
```

Comment:
Everything LGTM,

Now explore more on ChangeNotifierProvider and try to define success state, error state and loading state as well.

Basically to explain in brief, state management should also handle few scenarios. Let say that data that you wanted after increment , is not received for some reason right ? In that case the app should show an error test that data not received. And then there should be a loading state as well, so when that calculation is going on in background there should a loading in the screen to let people know that something is loading, similarly the success state is when you have the data with yourself.

So this PR looks good to me, no need to make any changes. Do one thing, make a different PR and now explore how to use API calls with state management and fix the recipe app with state management and send me the PR by end of this week . This is your new task. Keep in mind that it should have an error , success and loading state.